### DISCUSSION TITLE
could not decode rune

### DISCUSSION THREAD

**Discussion by mschneider82** [charmbracelet/bubbletea#664] [Q&A]
i am using the mouse input and always getting a program exit after a while because of "could not decode rune"

I would expect it continues instead of failing on a invalid utf 8 rune.

```
		r, width := utf8.DecodeRune(b[i:])
		if r == utf8.RuneError {
			return nil, errors.New("could not decode rune")
		}
```

**Comment by CalebJohnHunt** [comment]
Could you provide more code? Like your model, `View`, `Update`, and `Init` functions? If you have time, could you provide a [minimum reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)?

**Comment by mschneider82** [comment]
@bashbunni - its an issue in the lib not a discussion, the lib bubbletea returns an error which will make the programm exit. 
the better solution should be to skip and ignore runeerrors instead of error return.

**Reply by muesli** [reply]
I agree, that's an issue.
